include/libdnf/rpm/transaction.hpp

```diff
@@ -274,6 +275,45 @@ class Transaction {
     /// @param item  item to be erased
     void erase(TransactionItem & item);
 
+    /// Fill the RPM transaction from transaction packages.
+    /// @param transcation_packages The transaction packages to add.
+    /// @param rpm_transcation_items_out A vector of RPM transaction items of type T which will be filled up.
+    ///                                  TODO(lukash) temporary, needs more code moving around to simplify.
+    template<typename T>
+    void fill_transaction(
+        const std::vector<libdnf::base::TransactionPackage> & transaction_packages,
+        std::vector<std::unique_ptr<T>> & rpm_transaction_items_out) {
+        for (auto & tspkg : transaction_packages) {
+            auto fiddle_the_ptr = [&rpm_transaction_items_out](const libdnf::base::TransactionPackage & tspkg) {
```

Comment:
I'm not fond of the mechanism used, it was copy-pasted in each of the switch cases below, so I at least put it into the lambda. And tbh. I consider the name of the lambda quite fitting :nerd_face: I'm kinda hoping this will be improved (as the TODO in the method docstring says), but it may have to stay...

I'll add an explanatory comment.